### ISSUE TITLE
Accessibility updates needed

### ISSUE THREAD

**Issue by maggiewachs** [filamentgroup/fg-carousel#1]
Recs based on the [WAI ARIA authoring practices for carousel](https://www.w3.org/TR/wai-aria-practices-1.2/#carousel): 

* each slide should be focusable & accessed with a single tab stop, especially if it can contain other pieces of content (i.e., heading or caption, links, or if the entire thing can be linked). Managed focus _could_ be used on slides that are just images w/ no interactive elements, but if content can vary across carousels in a single site/app, it's best to avoid.
* user controls for stopping auto-play, moving to next/prev slide should be in the source before the carousel itself to make keyboard/SR users aware they exist
* the slide nav should follow the tab pattern (managed focus)
* our demos need a clearer focus state for the tab nav (sometimes the top side of the outline is cut off and it's less noticeable)

Things to consider: 
* what's needed for a simple scrolling region (like a filmstrip or  image reel) vs more full featured carousel (like retail ads on a home page)
* can next/prev buttons live in the markup at all times and use `aria-disabled` instead of relying on JS to append?
* use anchor links? (we'd still need to layer on managed focus for tab nav, but this could be a nice markup pattern to start with)

**Comment by scottjehl** [comment]
Seems we agreed so far that the next/prev links, when disabled, should have `aria-disabled` instead of `disabled` so that they're always in the a11y tree even when disabled

**Comment by maggiewachs** [comment]
Noting things here as they come up: 

* it would help to add the total number of images to the group label and/or add which slide you're on in the list (1 of 5) or whether it's first or last, which would help make sense of the next/prev disabled state
    * if you switch to a UL for the slides, most screen readers will announce it as a list of N items
* "prev" button needs an aria-label with the complete word. If it doesn't end up sounding too repetitive, might help to have both labeled as "previous slide" and "next slide"
* I'm debating whether the thumb nav is useful for screen readers in its current location after the slides. Once you've gone through the slides it's redundant unless you want to use it to jump to a specific slide. 
* if we keep thumb nav for SRs, thumbnail nav links need accessible names, either via alt text on the image, sr-only text, or aria-label. 
* ideally the thumb nav would be a single tab stop w/ managed focus. Rn it's a little cumbersome to tab through slides & tab through corresponding nav
* the tab order should remain consistent across carousel variants. In the img-only example, the order is container > prev/next > nav. In the captioned example, it's container > _each slide_ > prev/next > nav. Suggested order: 
    * container > prev/next (and pause, if relevant) > each slide > nav -- this way when using the dot nav users can shift+Tab directly back to the slide container (instead of the prev/next buttons)
* in the dot nav, visible focus state should be present at all times (rn it disappears) 
* for the img + caption example, VO doesn't associate the image with the caption/link, they're treated as separate items in the group (img > link > img > link...). Assuming the relationship matters, they should be grouped in the markup (img in link?).

**Comment by maggiewachs** [comment]
@scottjehl I tested the first 2 examples in the demos (Standard & Dots), and found some things that need to be fixed. I sorted them into compliance issues (need to be addressed to clear WCAG 2.1) vs usability improvements.

Compliance:

* Nav thumbnails need accessible names (they all say "link, demo")
* Prev button needs a label (reads "P R E V")
* When using arrows to scroll or prev/next buttons there's no audible feedback on the carousel change
* When you use Enter key to select a slide with a thumbnail/dot, there's no audible feedback on the carousel change (or visible focus shift to that slide; focus remains on the thumb nav option)
* Visible focus state on the dot nav appears and disappears
* Using arrow keys in the thumb/dot nav moves the slides & corresponding blue highlight, but not the focus state. While it looks like you're choosing a new slide w/ arrows (managed focus), you're only moving the highlight; when you hit Enter you go back to the last focused thumbnail (tab stops for navigation). 

Usability: 

* You have to Tab through the entire list to reach the prev/next buttons, and then you Tab from there to the thumb nav. It's unclear how the prev/next buttons are useful to keyboard & SR users. 
* Slide contents are not grouped, so each image and link is read as separate content items.
